cpp/arcticdb/async/async_store.hpp

```diff
@@ -8,187 +8,196 @@
 #pragma once
 
 #include <arcticdb/async/task_scheduler.hpp>
-#include <arcticdb/util/clock.hpp>
-#include <arcticdb/storage/store.hpp>
+#include <arcticdb/async/tasks.hpp>
 #include <arcticdb/entity/performance_tracing.hpp>
+#include <arcticdb/processing/clause.hpp>
 #include <arcticdb/storage/library.hpp>
-#include <folly/futures/Future.h>
-#include <arcticdb/async/tasks.hpp>
+#include <arcticdb/storage/store.hpp>
 #include <arcticdb/stream/stream_utils.hpp>
-#include <arcticdb/processing/clause.hpp>
+#include <arcticdb/util/clock.hpp>
+#include <folly/futures/Future.h>
 
 namespace arcticdb::async {
 
-template<class ClockType = util::SysClock>
-class AsyncStore : public Store {
-public:
-    AsyncStore(
-        std::shared_ptr<storage::Library> library,
-        const arcticdb::proto::encoding::VariantCodec &codec) :
-        library_(std::move(library)),
-        codec_(new arcticdb::proto::encoding::VariantCodec{codec}) {
+template <class ClockType = util::SysClock> class AsyncStore : public Store {
+  public:
+    AsyncStore(std::shared_ptr<storage::Library> library,
+               const arcticdb::proto::encoding::VariantCodec &codec)
```

Comment:
Is it though `int* a, b;`

Reply:
I think we never use that style because of said confusion. Delcaring a reference and a non-referene variable on the same line is looking for trouble anyway.

I would consider that an unwelcome left-over from C.

Reply:
Agreed, most static analyzers would flag this because of the known ambiguity